src/Ninject.Web.AspNetCore.Test/ServiceProviderReference/ServiceProviderReferenceTest.cs

```diff
@@ -81,6 +81,30 @@ public void ServiceProvider_SingletonService_CreatesSingleInstance()
 			instances.Aggregate((a, b) => a == b ? b : null).Should().NotBeNull();
 		}
 
+		[Fact]
+		public void ServiceProvider_ScopedServiceResolveServiceProvider_ReturnsScopedServiceProvider()
+		{
+			// Interesting side-note here: The ServiceProvider that is created is not actually the same object as when
+			// the IServiceProvider is requested from that provider.
+			var provider = CreateServiceProvider();
+
+			var rootProvider = provider.GetService<IServiceProvider>();
+			// Yes, requesting IServiceProvider will actually return a ServiceProviderEngineScope which is the root
```

Comment:
And yes, the `provider` and `rootProvider` represent the same thing but they are different objects 🤷‍♂️